tests/unit/test_sagemaker_steps.py

```diff
@@ -482,6 +482,135 @@ def test_training_step_creation_with_framework(tensorflow_estimator):
         'End': True
     }
 
+@patch('botocore.client.BaseClient._make_api_call', new=mock_boto_api_call)
+@patch.object(boto3.session.Session, 'region_name', 'us-east-1')
+def training_step_merges_hyperparameters_from_constructor_and_estimator(tensorflow_estimator):
```

Comment:
Just unit tests seem sufficient to me. This mostly has to do with static ASL generation and doesn't really rely on execution.

Reply:
yeah i also deferred writing tests because it already takes long as it is and we are validating against the generated definition